src/workloads/scale/MultiPlanStorm.yml

```diff
@@ -0,0 +1,93 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: >
+  The workload tests the server under a "multi-plan storm" situation. The same query requiring a
+  multi-plan is executed by many threads, each of them triggering a multi-plan. Each of the plans in
+  the multi-plan, on each thread, is index scanning a large number of documents, while maintaining a
+  RecordId set. This causes the memory footprint to increase until the server is eventually OOM
+  killed.
+
+GlobalDefaults:
+  MaxPhases: &MaxPhases 2
+  # The amount of documents here is important, since trial-runs in the query planner "tournament"
+  # will only use ~30% of total document count increasing the number of total documents helps
+  # reproduce the issue.
+  NumDocuments: &NumDocuments 30000000
+  PerfTagDocs: &PerfTagDocs 300
+  RandomTagDocs: &RandomTagDocs
+    ^NumExpr:
+      withExpression: "num_docs - num_perf_tags"
+      andValues: {num_docs: *NumDocuments, num_perf_tags: *PerfTagDocs}
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 6000
```

Comment:
Interesting, did you choose this number intentionally? What determines choosing maxPoolSize over number of threads below?

Reply:
Left over from attempting running 10M docs with 5K threads. Will bring it down. We need as many connections as threads + 3 I think it was? Can't remember exactly what for genny was using those extra connections though.